routes/subscriber.js

```diff
@@ -0,0 +1,17 @@
+
+
+function substituteReferenceData(req, res) {
+    const {referenceData} = req.body;
+    delete req.body.referenceData;
+    const response = interpolate(JSON.stringify(req.body.payload), referenceData); // To substitute ref data into payload
+    res.status(200).json(JSON.parse(response));
+}
+
+function interpolate(template, params) {
+    for (let key in params) {
+        template = template.replace(new RegExp('\\{' + key + '\\}', 'g'), params[key])
```

Comment:
@ishani3030 - can you check the system health, if the payload size was more than 10 MB?

to check system health,
- On windows, you can use task manager.
- On Linux, you can use top command.

Reply:
Yes, I have tried to execute same with payload around 10mb. CPU utilization gets high up to 75-80%